### ISSUE TITLE
Failed to create game ( Long Game wait ) What Am I doing Wrong?

### ISSUE THREAD

**Issue by Apotrs** [kolton/d2bot-with-kolbot#2819]
So I've been botting for a while and it seems like my bot usually stops the game being made then restarting.

Sometimes the bot will be successful in making the game but then 10+ more waiting in line with cancelling to create the next game successfully, What's going on?
![ftj d2](https://user-images.githubusercontent.com/71225288/97493661-4bc28d80-1933-11eb-9da9-7574ed711742.jpg)
![ftj d3](https://user-images.githubusercontent.com/71225288/97493665-4cf3ba80-1933-11eb-95c4-de6b13d085ae.jpg)
![Ftj D](https://user-images.githubusercontent.com/71225288/97493667-4ebd7e00-1933-11eb-881b-89b6ba2dbc91.jpg)

More fails

![more d22](https://user-images.githubusercontent.com/71225288/97498414-21280300-193a-11eb-8879-384e91262c63.jpg)
![more d2](https://user-images.githubusercontent.com/71225288/97498416-21c09980-193a-11eb-9ac2-9f036968e2ad.jpg)

FINALLY A GAME WAS MADE!?!?
![finally](https://user-images.githubusercontent.com/71225288/97500186-29357200-193d-11eb-85b7-ef1f7d47ee19.jpg)

**Comment by Karakachanov** [comment]
Your not doing anything wrong. Many people have the same issue. It's Server side , Blizz made something, I was in game when they restarted the server about a week ago and since that restart it's been acting like that occasionally putting you in waiting ques when creating games, and if the que is longer than what the bot can handle it attempts to create new game ... if this happen multiple times you may get temp ban (which happened to me already, i've got 3 hours ban)

**Comment by AbbyRoad** [comment]
I agree that you have done nothing wrong. I have had similar issues.  In the last 18 hours I have failed to make a single game when botting. In all cases the bot bailed on the que well before the "WaitInLineTimeout" limit was reached. (Just as you have noted above.)  In watching the bot what I noticed is that if the que countdown stalls at some random point for too long, the bot leaves the que.  So I suspect there is a coding issue with the bot.  i.e. it will leave the que if either (1) the max wait time has been hit; or (2) if the (instantaneous?) pace of the wait in line que countdown falls below a specific level.

As another test, yesterday  I tried to make a few games manually, the games were always made.  However, once I restarted the bot, the games were never successfully made.

So, I believe there is glitch in the bot's code.

With all that said, later today I'm going to change the lead bot and see if that is a solution. If you look through the previous issues you will see that seemed to have helped with this wait in line issue in the past.

**Comment by qnts** [comment]
The exact problem I encountered days ago. However I was able to solve it yesterday.
Try this: https://gist.github.com/qnts/c8515e69b93e0653a06ab2356513e80c#gistcomment-3528694

**Comment by AbbyRoad** [comment]
Thanks qnts

I copied the sections of code you highlighted in your comments section, and I was able to get 1 out of 10 attempts to successfully run.  I did increase the WaitInLineStuckingTimeout from 30 to 600 sec, so I am pretty confident that I did not hit that timeout limit.

Next, I over-wrote my entire d2botleader file with yours, but no real change in the results, other than I seem to stay in the queues (on average) longer before I fail.  I'm on US East softcore ladder.

Regards;

AbbyRoad

**Comment by AbbyRoad** [comment]
Hi qnts

In the last 12 attempts, I have had zero successful games.  I finally had the opportunity to sit down and watch what was happening.  During the wait in line, typically it would take 1 - 2 seconds between refreshes of the leader position in the line.  At one step, the refresh time increased to about 10 secs, but the refresh of the position in  line finally occurred.

Later the refresh time increased to about 20 seconds, before the "Failed to create game (queue)" message came up.

I had the Leader's Create Game screen in the foreground, and just prior to both of the extended refresh times I heard a noise (like you would hear if you hit the enter key when you are on the make game screen) and the Leader's Create Game screen flickered.  Do you any idea why the program would be sending additional "enter" commands while in the queue? 

Regards;

AbbyRoad

**Comment by qnts** [comment]
I has 10k queue and it takes 20-30 min in queue to create a game. Before the modification, my bot cannot create even a single game due to the premature canceling.
I think it's either blizz server or your network problem. I'm on Asian ladder FYI.

**Comment by AbbyRoad** [comment]
Thanks.  Our queue is "only" 4k, but it takes 40 - 50+ minutes to create a game.  You should see if you can get your modified code included in the official kolbot folder.

Regards;

AbbyRoad

**Comment by qnts** [comment]
@AbbyRoad getting code merging is a long procedure though. I got another enhancement for the leader script and will open it to the public soon. Also you should join the discord development channel for further fixes and updates by the community.
